drivers/usb/udc/udc_dwc2.c

```diff
@@ -585,9 +585,21 @@ static void dwc2_handle_xfer_next(const struct device *dev,
 	if (USB_EP_DIR_IS_OUT(cfg->addr)) {
 		dwc2_prep_rx(dev, buf, cfg);
 	} else {
-		if (dwc2_tx_fifo_write(dev, cfg, buf)) {
-			LOG_ERR("Failed to start write to TX FIFO, ep 0x%02x",
-				cfg->addr);
+		int err = dwc2_tx_fifo_write(dev, cfg, buf);
+
+		if (err) {
+			if (err == -ENOENT) {
+				LOG_WRN("Cannot write to TX FIFO, ep 0x%02x inactive", cfg->addr);
+			} else {
+				LOG_ERR("Failed to start write to TX FIFO, ep 0x%02x (err: %d)",
+					cfg->addr, err);
+			}
```

Comment:
@jfischer-no, does the following change fully address your [comment](https://github.com/zephyrproject-rtos/zephyr/pull/77685#discussion_r1742031143)?

```suggestion
			LOG_ERR("Failed to start write to TX FIFO, ep 0x%02x (err: %d)",
				cfg->addr, err);
```

Reply:
I would be happy to see this changed to:
```c
		if (err) {
			LOG_ERR("Failed to start write to TX FIFO, ep 0x%02x (err: %d)",
				cfg->addr, err)

			buf = udc_buf_get(dev, cfg->addr)
			if (udc_submit_ep_event(dev, buf, -ECONNREFUSED)) {
				LOG_ERR("Failed to submit endpoint event");
			};

			return;

```

Reply:
aligned according to your suggestions and added missing semicolons